src/main.rs

```diff
@@ -0,0 +1,169 @@
+    println!("::Q1::");
+    let mut answer_stream = transaction.query().match_(q.as_str()).unwrap();
+    while let Some(result) = answer_stream.next().await{
+        match result {
+            Ok(concept_map) => {
+                for (_, concept) in concept_map {
+                    if let Concept::Attribute(Attribute { value: Value::String(value), .. }) = concept {
+                        println!("{}",value);
+                    }
+                }
+            }
+            Err(err) => {
+                panic!("An error occurred fetching answers of a Match query: {err}")
+            }
+        }
+    }
+    Ok(())
+}
+async fn query2(connection: Connection,q: String)->std::io::Result<()>{    
+    let databases = DatabaseManager::new(connection.clone());
+    let session = Session::new(databases.get(TEST_DATABASE).await.unwrap(), Data).await.unwrap();
+    println!("::Q2::");
+    let transaction = session.transaction(Read).await.unwrap();
+    let answer = transaction.query().match_aggregate(q.as_str()).await.unwrap();
+    println!("Answe for Q2:  {}\n",answer.into_i64());
+    Ok(())
+}
+async fn query3(connection: Connection,q: String)->std::io::Result<()>{    
+    let databases = DatabaseManager::new(connection.clone());
+    let session = Session::new(databases.get(TEST_DATABASE).await.unwrap(), Data).await.unwrap();
+    let transaction = session.transaction(Read).await.unwrap();
+    println!("::Q3::");
+    let mut answer_stream = transaction.query().match_(q.as_str()).unwrap();
+    while let Some(result) = answer_stream.next().await{
+        match result {
+            Ok(concept_map) => {
+                for (_, concept) in concept_map {
+                    if let Concept::Attribute(Attribute { value: Value::String(value), .. }) = concept {
+                        println!("{}",value);
+                    }
+                }
+            }
+            Err(err) => {
+                panic!("An error occurred fetching answers of a Match query: {err}")
+            }
+        }
+    }
+    Ok(())
+}
+
+async fn runIO(connection: Connection){
+    println!("\nQ1->What places could buy raw_food=$x ?\nQ2->Get count of non-vegetarian outlets with vegetarian specialities.\nQ3->Get count raw items sold at places with avg_rating more tha $r and has a dish using it as raw_ingredient with price greater than $p units.\nWhat query would you like to make? Enter 1,2 or 3.\n");
+    let qtype: i32=read!();
+    if qtype==1{
+        println!("Enter the name of raw_food x : ");
+        let inp: String=read!();
+        let x=inp.as_str();
+        let p0="match $rf isa raw_food, has name \"";
+        let p1="\";
+        $is_i (raw_food: $rf,dish: $d) isa is_ingredient;
+        $sl (seller: $m,product: $d) isa sells;
+        $m has name $n,has call_number $c;
+        get $n;";
+        let mut f="".to_owned();
+        f=f+p0+x+p1;
+        query1(connection,f).await.unwrap();
```

Comment:
Much of the code inside this if block is specific to `query1`. This function would read nicer if we leave everything that doesn't have to do with collecting the input from the user and putting it into the relevant query function.

Reply:
The same goes for the other two queries.